pkg/v1/remote/transport/basic.go

```diff
@@ -57,6 +71,19 @@ func (bt *basicTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 				in.Header.Set("Authorization", hdr)
 			}
 		}
+		resp, err := bt.inner.RoundTrip(in)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusForbidden {
+			if last {
+				return resp, nil
+			}
+			continue
```

Comment:
We probably want to debug-log each failure we get, in case it's helpful. 

https://pkg.go.dev/github.com/google/go-containerregistry/pkg/logs

Reply:
fixed